src/main/java/org/broadinstitute/hellbender/tools/diagnostics/PrintBGZFBlockInformation.java

```diff
@@ -84,16 +97,57 @@ protected void onStartup() {
+        boolean sawNonFinalTerminatorBlock = false;
+
         try ( InputStream bgzfInputStream = Files.newInputStream(bgzfPath) ) {
-            BGZFBlockMetadata blockInfo = null;
+            outStream.printf("BGZF block information for file: %s\n\n", bgzfPath.getFileName());
+
+            BGZFBlockMetadata blockInfo;
+
             while ( (blockInfo = processNextBlock(bgzfInputStream, bgzfPathString)) != null ) {
-                outStream.println(String.format("Block at file offset %d", blockInfo.blockOffset));
-                outStream.println(String.format("\t- compressed size: %d", blockInfo.compressedSize));
-                outStream.println(String.format("\t- uncompressed size: %d", blockInfo.uncompressedSize));
+
+                // If we saw a 0-byte terminator block that was not the final block in the file,
+                // emit a warning
+                if ( previousBlockInfo != null && previousBlockInfo.uncompressedSize == 0 ) {
+                    sawNonFinalTerminatorBlock = true;
+
+                    outStream.println("*************************************************************************");
+                    outStream.println("WARNING: Premature BGZF 0-byte terminator block found before final block!");
+                    outStream.println("*************************************************************************");
+                    outStream.println();
+                }
+
+                outStream.printf("Block at file offset %d\n", blockInfo.blockOffset);
+                outStream.printf("\t- compressed size: %d\n", blockInfo.compressedSize);
+                outStream.printf("\t- uncompressed size: %d\n", blockInfo.uncompressedSize);
                 outStream.println();
+
+                previousBlockInfo = blockInfo;
             }
         } catch ( IOException e ) {
-            throw new UserException("Error while parsing BGZF file. Error message was: " + e.getMessage(), e);
+            throw new UserException.CouldNotReadInputFile("Error while parsing BGZF file.", e);
+        }
+
+        // Check whether the last block in the file was a 0-byte BGZF terminator block
+        if ( previousBlockInfo == null || previousBlockInfo.uncompressedSize != 0 ) {
+            outStream.println("********************************************************");
+            outStream.println("WARNING: Final BGZF 0-byte terminator block was MISSING!");
+            outStream.println("********************************************************");
+            outStream.println();
+        } else {
+            outStream.println("*****************************************");
+            outStream.println("Final BGZF 0-byte terminator block FOUND!");
+            outStream.println("*****************************************");
+            outStream.println();
+        }
+
+        // Emit a warning at the end if we encountered any terminator blocks before the final block:
+        if ( sawNonFinalTerminatorBlock ) {
```

Comment:
I would almost think that this should also print out the location of the first one found, just so its use friendly and doesn't confuse anybody about there being two empty blocks in a row at the end.

Reply:
Either that or clearly separate this from the very similarly formatted output line that appears right before it in the output.

Reply:
Done -- I added block numbers, and reference them in the error messages.